compaction/compaction_manager.cc

```diff
@@ -926,11 +926,17 @@ void compaction_manager::submit(replica::table* t) {
 }
 
 class compaction_manager::offstrategy_compaction_task : public compaction_manager::task {
+    std::vector<sstables::shared_sstable> _old_sstables;
+
 public:
     offstrategy_compaction_task(compaction_manager& mgr, replica::table* t)
         : task(mgr, t, sstables::compaction_type::Reshape, "Offstrategy compaction")
     {}
```

Comment:
I am thinking we could make it simpler by setting _old_sstables in offstrategy_compaction_task ctor above.

: task(mgr, t, sstables::compaction_type::Reshape, "Offstrategy compaction")
, _old_sstables(t.maintenance_sstable_set())

and potentially use const qualifier as we don't want to change it.

Reply:
See below.  This approach definitely cross my mind, but I realized we currently get the maintenance set after
acquiring the off strategy semaphore, and I wonder if it might possibly change if we block.
If it might, then we should make the decision only after acquiring the semaphore like it is done in this version.